packages/plugin-expo-device/device.js

```diff
@@ -20,10 +22,22 @@ module.exports = {
     // get the initial orientation
     updateOrientation()
 
+    // Fetch the free disk space up front because it's an async API so we don't
+    // want to do this in the onError callback. This means the reported free
```

Comment:
We _can_ do async `onError` callbacks, but we strongly advise against it. But I think the disk space could quite conceivably fill up after the app has started, so either we need to convert the `onError` and `onSession` callbacks to be `async` OR set an interval to poll the free disk space while the app is running. I'm not sure what a good interval for that timeout would be though.

Reply:
Yeah I guess any interval we pick would still be wrong if the thing causing the crash also writes something large to disk

Is this better to leave out for now to avoid inaccurate data/making the `onError` async?

From a look at the source, it doesn't _seem_ like a super expensive API so a low interval may be ok, but that might be deceiving